src/core/matter/mod.rs

```diff
@@ -1,17 +1,102 @@
-pub(crate) mod tables;
+#[cfg(feature = "python")]
+use pyo3::{
+    exceptions::{PyException, PyValueError},
+    prelude::*,
+};
 
 use base64::{engine::general_purpose as b64_engine, Engine};
 
 use crate::core::util;
 use crate::error::{err, Error, Result};
 
+pub mod tables;
+
+#[cfg_attr(feature = "python", pyclass)]
 #[derive(Debug, Clone)]
 pub struct Matter {
     pub(crate) raw: Vec<u8>,
     pub(crate) code: String,
     pub(crate) size: u32,
 }
 
+#[cfg(feature = "python")]
+#[pymethods]
+impl Matter {
+    #[new]
+    fn py_new(
+        code: Option<&str>,
+        raw: Option<&[u8]>,
+        raw_size: Option<usize>,
+        qb64: Option<&str>,
+        qb64b: Option<&[u8]>,
+        qb2: Option<&[u8]>,
+        strip: Option<bool>,
+    ) -> PyResult<Self> {
+        let result = if let Some(code) = code {
+            let (raw, raw_size) = if raw.is_none() || raw_size.is_none() {
+                return Err(PyValueError::new_err("code present, raw and raw_size missing"));
+            } else {
+                (raw.unwrap(), raw_size.unwrap())
+            };
+
+            Self::new_with_code_and_raw(code, raw, raw_size)
+        } else if let Some(qb64) = qb64 {
+            Self::new_with_qb64(qb64)
+        } else if let Some(qb64b) = qb64b {
+            Self::new_with_qb64b(qb64b)
+        } else if let Some(qb2) = qb2 {
+            Self::new_with_qb2(qb2)
+        } else {
```

Comment:
Someone should add validation if this gets merged, that multiple things aren't set. Example: qb2 and qb64 both specified = bad.

Reply:
This comment is only outdated becode code moved, we should still do this.